sensor/common/scannerclient/grpc_client.go

```diff
@@ -83,10 +176,68 @@ func (c *Client) GetImageAnalysis(ctx context.Context, image *storage.Image, cfg
 
 	log.Debugf("Received image components from local Scanner for image %s", name)
 
-	return resp, nil
+	return &ImageAnalysis{
+		ScanStatus:   resp.GetStatus(),
+		ScanNotes:    resp.GetNotes(),
+		V1Components: resp.GetComponents(),
+	}, nil
+}
+
+// Close closes and cleanup the client connection.
+func (c *v2Client) Close() error {
+	return c.conn.Close()
+}
+
+func convertIndexReportToAnalysis(ir *v4.IndexReport) *ImageAnalysis {
+	var st scannerV1.ScanStatus
+	switch ir.GetState() {
+	case "Terminal", "IndexError":
+		st = scannerV1.ScanStatus_FAILED
+	case "IndexFinished":
+		st = scannerV1.ScanStatus_SUCCEEDED
+	default:
+		st = scannerV1.ScanStatus_ANALYZING
+	}
+	return &ImageAnalysis{
+		ScanStatus: st,
+		V4Contents: ir.GetContents(),
+	}
+}
+
+func (c *v4Client) GetImageAnalysis(ctx context.Context, image *storage.Image, cfg *types.Config) (*ImageAnalysis, error) {
+	hid := fmt.Sprintf("/v4/containerimage/%s", image.GetId())
```

Comment:
I thought hid needs to be image digest?

Reply:
I thought `image.GetId()` **is the image digest**. Is it like it can the image ID can be a tag?

Reply:
I see image id has been used this way in the codebase so I assume it is a valid digest:
```
err := arrayWriter.WriteObject(&splunkDeploymentEvent{
					Type:        "deployment",
					Cluster:     deployment.GetClusterName(),
					Namespace:   deployment.GetNamespace(),
					Labels:      deployment.GetLabels(),
					Annotations: deployment.GetAnnotations(),
					Deployment:  deployment.GetName(),
					ImageDigest: c.GetImage().GetId(),
				})
```
In Clair integration, I also notice @RTann get digest this way:
```
ccDigest, err := claircore.ParseDigest(imageutils.GetSHA(image))
if err != nil {
	return nil, errors.Wrapf(err, "Clair v4: parsing image digest for image %s", imgName)
}
digest := ccDigest.String()
```
I am not sure if this works better with Clair

Reply:
e318e794